scripts/upload-housing.js

```diff
@@ -0,0 +1,252 @@
+    }
+
+    if (path.extname(filePath).toLowerCase() !== `.${extension}`) {
+      warnAndExit(`ERROR! Expected a .${extension} file: ${filePath}`);
+    }
+  });
+
+  process.env.GOOGLE_APPLICATION_CREDENTIALS = "serviceAccount.json";
+  const app = initializeApp();
+  const db = getFirestore(app);
+
+  // Check if the collection exists
+  const collections = (await db.listCollections()).map((c) => c.id);
+  if (!collections.includes("spotlights")) {
+    warnAndExit(`ERROR! housing must match an existing collection id: ${collections.join(
+      ", "
+    )}.
+     `);
+  }
+
+  const docRef = db.collection("spotlights").doc("housing");
+
+  const dir = `${localDir}/spotlights/housing`;
+
+  if (localDir) {
+    if (fs.existsSync(dir)) {
+      if (overwrite) {
+        console.warn(
+          `WARNING! Directory exists locally. Overwriting... ${dir}`
+        );
+      } else {
+        warnAndExit(
+          `ERROR!: Directory exists locally. Use the overwrite flag if you wish to continue: ${dir}`
+        );
+      }
+    }
+  } else {
+    // Check if the document exists
+    const docSnapshot = await docRef.get();
+    if (docSnapshot.exists) {
+      if (overwrite) {
+        console.warn(
+          `WARNING! Document exists in Firestore. Overwriting... housing`
+        );
+      } else {
+        warnAndExit(
+          `ERROR!: Document exists in Firestore. Use the overwrite flag if you wish to continue: housing`
+        );
+      }
+    }
+  }
+
+  // read in the files to the data property on each file object
+  files.forEach((file) => {
+    const json = getJson(file.filePath);
+    if (file.property) {
+      file.data = json(file.property);
+    } else {
+      file.data = json;
+    }
+
+    // check the file isArray
+    if (file.isArray !== Array.isArray(file.data)) {
+      warnAndExit(
+        `ERROR!: The file for ${file.field} at ${
+          file.filePath
+        } was expected to contain ${
+          file.isArray ? "an array" : "an object"
+        }, but it instead contained ${
+          Array.isArray(file.data) ? "an array" : "a " + typeof file.data
+        }.`
+      );
+    }
+
+    // check the schema
+    if (file.schema) {
+      // Check data as an array of objects
+      const data = file.isArray ? file.data : [file.data];
+
+      data.forEach((row) => {
+        checkSchema(file.schema, row, file.filepath);
+      });
+    }
+  });
+
+  if (localDir) {
+    // Make directory, no harm done if already exists
+    fs.mkdir(dir, { recursive: true }, (err) => {
+      if (err) {
+        warnAndExit(err);
+      }
+
+      files.forEach(({ field, data }) => {
+        const path = `${dir}/${field}.json`;
+
+        fs.writeFile(path, JSON.stringify(data), (err) => {
+          if (err) {
+            warnAndExit(err);
+          }
+          console.log(`SUCCESS! Created file: ${path}`);
+        });
+      });
+    });
+  } else {
+    await docRef.set({
+      // TO_RVIEW docRef.select is not a function
```

Comment:
delete this line?

Reply:
Ahh, I misspelled "Review and missed it!"